File: C++/BOJ/11725.cpp
#include<iostream>
#include<vector>
#include<deque>
using namespace std;

void    bfs(vector<vector<int>> graph, vector<int> ans)
{
    int current;
    deque<int> queue;
    queue.push_back(1);
    while(queue.size())
    {
        current = queue.front();
        queue.pop_front();
        for (auto num : graph[current])
            if (ans[num] == 0)
            {
                ans[num] = current;
                queue.push_back(num);
            }
    }
    for (int i = 2; i < ans.size(); i++)
        cout<<ans[i]<<"\n";
}

int main()
{
    int n;
    int left, right;
    vector<vector<int>> graph;
    vector<int> ans;
    cin>>n;
    for (int i = 0; i <= n; i++)
    {
        graph.push_back({});
        ans.push_back(0);
    }
    for (int i = 1; i < n; i++)
    {
        cin>>left>>right;
        graph[left].push_back(right);
        graph[right].push_back(left);
    }
    bfs(graph, ans);

}

File: DFS_BFS/11725.py
import sys
from collections import deque

n = int(sys.stdin.readline())
graph = [[] for i in range(n + 1)]
for i in range(n - 1):
    left, right = map(int, sys.stdin.readline().split())
    graph[left].append(right)
    graph[right].append(left)

queue = deque()
queue.append(1)
ans = [0]*(n+1)

def bfs():
    while queue:
        current = queue.popleft()
        for num in graph[current]:
            if ans[num] == 0:
                ans[num] = current
                queue.append(num)
bfs()
for i in range(2, len(ans)):
    print(ans[i])
